File: php/difference-of-squares/difference-of-squares.php
<?php

function squareOfSums($num)
{
	$numberToSquare = 0;
	for ( $i = 1; $i <= $num; $i++ ) {
		$numberToSquare += $i;
	}

	return pow($numberToSquare, 2);
}

function sumOfSquares($num)
{
	$totalSquared = 0;
	for ( $i = 1; $i <= $num; $i++ ) {
		$totalSquared += pow($i, 2);
	}
	return $totalSquared;
}

function difference($num)
{
	return squareOfSums($num) - sumOfSquares($num);
}

File: ruby/difference-of-squares/difference_of_squares.rb
# Squares class
class Squares
  def initialize(number)
    @number = number
  end

  def square_of_sums
    total = @number
    @number.times do |i|
      total += i
    end
    total**2
  end

  def sum_of_squares
    total = 0
    @number.times do |i|
      total += (i + 1)**2
    end
    total
  end

  def difference
    square_of_sums - sum_of_squares
  end
end
